### PR TITLE
Streaming data tests

### PR DESCRIPTION
**Pull request by dakinggg**
# What does this PR do?
Adds tests for composer + streaming for NLP datasets

TODO:
- [ ] figure out where these tests should actually read from (Q for Mihir)
- [x] unxfail all the scenarios as streaming bugs are fixed
- [x] mark as daily and remote
- [x] run locally on mac and gpu instance for world size 1/2 (ran `make test`, `make test-dist`, `WORLD_SIZE=2 make test-dist`, `make test-gpu`, `make test-dist-gpu`, and `WORLD_SIZE=2 make test-dist-gpu` on my Mac (excluding the gpu ones), and on a GPU instance)
- [x] In order to work on mac, this needs https://github.com/mosaicml/streaming/pull/97

# What issue(s) does this change relate to?
Closes [CO-1479](https://mosaicml.atlassian.net/browse/CO-1479)

# Before submitting
- [ ] Have you read the [contributor guidelines](https://github.com/mosaicml/composer/blob/dev/CONTRIBUTING.md)?
- [ ] Is this change a documentation change or typo fix? If so, skip the rest of this checklist.
- [ ] Was this change discussed/approved in a GitHub issue first? It is much more likely to be merged if so.
- [ ] Did you update any related docs and document your change?
- [ ] Did you update any related tests and add any new tests related to your change? (see [testing](https://github.com/mosaicml/composer/blob/dev/CONTRIBUTING.md#running-tests))
- [ ] Did you run the tests locally to make sure they pass?
- [ ] Did you run `pre-commit` on your change? (see the `pre-commit` section of [prerequisites](https://github.com/mosaicml/composer/blob/dev/CONTRIBUTING.md#prerequisites))

[CO-1479]: https://mosaicml.atlassian.net/browse/CO-1479?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: tests/fixtures/autouse_fixtures.py
Comment (mvpatel2000) [inline tests/fixtures/autouse_fixtures.py:18]: What does this do? can you maybe leave a comment?

Comment (dakinggg) [reply tests/fixtures/autouse_fixtures.py]: added a comment. basically it just silences a warning from huggingface

File: tests/datasets/test_streaming_datasets_train.py
Comment (mvpatel2000) [inline tests/datasets/test_streaming_datasets_train.py:83]: Should `engine.close()` call shutdown on the workers?

**Review by mvpatel2000 — COMMENTED** [review]
LGTM, but want to hold off until after remote / daily migration to minimize moving pieces

File: tests/datasets/test_streaming_datasets_train.py
Comment (dakinggg) [reply tests/datasets/test_streaming_datasets_train.py:83]: probably. can't imagine a situation where the engine is closed but you don't want the dataloader workers shut down.

Comment (mvpatel2000) [reply tests/datasets/test_streaming_datasets_train.py:83]: Can we add it in this PR?

Comment (dakinggg) [reply tests/datasets/test_streaming_datasets_train.py:83]: sure, will add

Comment (dakinggg) [reply tests/datasets/test_streaming_datasets_train.py:83]: https://github.com/mosaicml/composer/pull/1842/commits/4919d5a3978ee6a163e590c23a83a2083b2c8012

**Review by mvpatel2000 — APPROVED** [review]
would like to get review from @nik-mosaic as well

File: tests/datasets/test_streaming_datasets_train.py
Comment (nik-mosaic) [inline tests/datasets/test_streaming_datasets_train.py:53]: This exact seed???

**Review by nik-mosaic — APPROVED** [review]
Looks good, with the assumption that any metrics tests will be separate --- this is simply a matter of testing that the calls to train.fit() succeed.

File: tests/datasets/test_streaming_datasets_train.py
Comment (nik-mosaic) [reply tests/datasets/test_streaming_datasets_train.py:53]: Resolving for now, but please follow up @dakinggg

Comment (dakinggg) [reply tests/datasets/test_streaming_datasets_train.py:53]: Not this exact seed. But the seeds needed to be unique across all parametrizations of this test. I think this has been fixed in the latest streaming, and I made a JIRA to clean up this test with the latest version of streaming

**Pull request merged**
